### DISCUSSION TITLE
Core needs updated. Issues with mysql connections

### DISCUSSION THREAD

**Discussion by rjstechnet** [WhiteCoreSim/WhiteCore-Dev#339] [General]
Just a note about mysql revisions requiring a bit of fixing for the projects. WhiteCoreSim included. OpenSim doesn't have the issue, but WhiteCoreSim isn't wanting to connect to later mysql instances. I do have it working on older mysql. I believe it has to do with the type of password mysql is looking for and the changes to mysql concerning this.

Also a question about how to add an abuse report. Not seeing it in the panel although the tables are there.

Thank you,
rjs

**Comment by rjstechnet** [comment]
Another question, is this project dead? I've contacted the developer several times with no responses. Are we still working on this?

Thank you,
rjs

**Comment by fly-man-** [comment]
Last I tested was with MariaDB and it connects without issues. MySQL 8.0 and higher will have issues 

Newer versions of MySQL force the use of creating the database user with a password different then the original implementation.

https://dev.mysql.com/doc/refman/8.0/en/upgrading-from-previous-series.html#upgrade-caching-sha2-password

**Comment by greythane** [comment]
I have tested the WhiteCore code with MySQL V8 quite a while ago and there were no observable problems then.  
If there have been changes since then then yes, there may be issues but I do not think they will be WhiteCore related rather than some updated requirement of MySQL 8.  

All instances I have used connect to a MariaDB server as I have found this to be more reliable

**Comment by rjstechnet** [comment]
Yeah, that's what I mentioned. I have it running on older mysql/mariadb. And yes, I know all about the changes and all to the way the passwords are handled. OpenSim spells it out in the conf files. :)

Any chance we can add the OpenSimulator scripting engine "Yengine" to WhiteCoreSim? Still prefer WhiteCoreSim since Aurora. Always liked the direction. I finally have some scripts that run on OpenSim that were done in 2006-2008. They finally work. :)

**Reply by fly-man-** [reply]
At the moment the Y-Engine is one of the things we have had on the list since 2019. The only downfall so far is that it would mean rewriting a large portion on the code which both me and @greythane haven't been able to do.

So if there is anyone that is willing to invest the time to implement it, please do

**Comment by rjstechnet** [comment]
I see fly-man-. It is alot of work and I do hope someone will read this and be able to help. It took years to get WhiteCoreSim where it is, but it has a way to go as well. At the moment, an admin can review abuse reports that a end user cannot create. Just things like this.

And then there are all the undone pieces in the panel. Under construction signs etc.. All that needs fixed for this really needs fixed. 
But at any rate, in my humble opinion, WhiteCoreSim "was" light years ahead at one point. Doing flat file based asset storage before anyone.
And utilizing the built in web servers as WhiteCoreSim does is just how it should be. You shouldn't have to run outside servers and have all the issues that come with xmlrpc. Oh, sorry xmlrpci. I'll report what I find, if it gets fixed it does, if it doesn't it doesn't. But this project and all the hard work over the years should not be left to bit rot. There we have it.

Thanks for all you have done over the years too fly-man-.  You too greythane!
